tests/test_patterns.py

```diff
@@ -0,0 +1,263 @@
+
+
+@pytest.mark.parametrize('given,expected', [
+    (']]K[[', ['K']),
+    (']]Bakemonogatari[[', ['Bakemonogatari']),
+    (']]Steins;Gate[[', ['Steins;Gate']),
+    (']]Kill la Kill[[', ['Kill la Kill']),
+    (']]ヒナまつり[[', ['ヒナまつり']),
+    (']]Re:Zero[[ ]]No Game No Life[[', ['Re:Zero', 'No Game No Life']),
+    ('', []),
+    (']][[', []),
+    (']] [[', []),
+    (']]  [[', []),
+    (']]trailing_whitespace [[', []),
+    (']] leading_whitespace[[', []),
+    (']] wrapping_whitespace [[', []),
+    ('no_whitespace]]before[[', []),
+    (']regular[', []),
+    (']]]]partial_before[[', []),
+    (']]partial_after[[[[', []),
+    (']]nested]]after[[[[', []),
+    (']]]]nested[[before[[', []),
+    (']]parens_after[[(', []),
+])
+def test_expanded_light_novel(given, expected):
+    actual = list(find_requests('light_novel', given, expanded=True))
+    assert actual == expected
+
+
+@pytest.mark.parametrize('given,expected', [
+    ('|K|', ['K']),
+    ('|Bakemonogatari|', ['Bakemonogatari']),
+    ('|Steins;Gate|', ['Steins;Gate']),
+    ('|Kill la Kill|', ['Kill la Kill']),
+    ('|ヒナまつり|', ['ヒナまつり']),
+    ('|Re:Zero| |No Game No Life|', ['Re:Zero', 'No Game No Life']),
+    ('', []),
+    ('||', []),
+    ('| |', []),
+    ('|  |', []),
+    ('|trailing_whitespace |', []),
+    ('| leading_whitespace|', []),
+    ('| wrapping_whitespace |', []),
+    ('no_whitespace|before|', []),
+    ('||extended||', []),
+    ('||partial_before|', []),
+    ('|partial_after||', []),
+    ('|nested|after||', []),
+    ('||nested|before|', []),
+    ('|parens_after|(', []),
+])
+def test_regular_visual_novel(given, expected):
+    actual = list(find_requests('visual_novel', given))
+    assert actual == expected
+
+
+@pytest.mark.parametrize('given,expected', [
+    ('||K||', ['K']),
+    ('||Bakemonogatari||', ['Bakemonogatari']),
+    ('||Steins;Gate||', ['Steins;Gate']),
+    ('||Kill la Kill||', ['Kill la Kill']),
+    ('||ヒナまつり||', ['ヒナまつり']),
+    ('||Re:Zero|| ||No Game No Life||', ['Re:Zero', 'No Game No Life']),
+    ('', []),
+    ('||||', []),
+    ('|| ||', []),
+    ('||  ||', []),
+    ('||trailing_whitespace ||', []),
+    ('|| leading_whitespace||', []),
+    ('|| wrapping_whitespace ||', []),
+    ('no_whitespace||before||', []),
+    ('|regular|', []),
+    ('||||partial_before||', []),
+    ('||partial_after||||', []),
+    ('||nested||after||||', []),
+    ('||||nested||before||', []),
+    ('||parens_after||(', []),
+])
+def test_expanded_visual_novel(given, expected):
+    actual = list(find_requests('visual_novel', given, expanded=True))
+    assert actual == expected
+
+
+def test_all_regular_patterns():
+    given = ('This is an **example** [Markdown] comment.\n'
+             'Here are some requests:\n'
+             '{anime 1} {anime2}\n'
+             '<manga 1> <manga2>\n'
+             ']light novel 1[ ]light_novel2[\n'
+             '|visual novel 1| |visual_novel2|\n\n'
+             '[Markdown]: https://commonmark.org/')
```

Comment:
Not the end of the world by any means, but it would be nice to include some link markdown here as well: `[This is a link.](https://google.com)`

Reply:
Ah. Yeah, that's what I was going for with the link reference
```markdown
This is an **example** [Markdown] comment.
...
[Markdown]: https://commonmark.org/
```
but it's easy enough to add an inline link instead. :grin: